### PR TITLE
Backport the in-memory functional tests to 2.1 

### LINKED ISSUE #2776: Test failure: UpgradedConnectionsCountsAgainstDifferentLimit
This test [fails](http://aspnetci/viewLog.html?buildId=524221&buildTypeId=Releases_21Public_UbuntuUniverse) occasionally with the following error:
```
System.IO.IOException : Expected connection close, received data instead: "HTTP/1.1 200 OK
Date: Mon, 06 Aug 2018 10:40:42 GMT
Content-Length: 0
"
   at Microsoft.AspNetCore.Testing.TaskExtensions.TimeoutAfter(Task task, TimeSpan timeout, String filePath, Int32 lineNumber)
   at Microsoft.AspNetCore.Server.Kestrel.FunctionalTests.ConnectionLimitTests.UpgradedConnectionsCountsAgainstDifferentLimit() in /_/test/Kestrel.FunctionalTests/ConnectionLimitTests.cs:line 89
--- End of stack trace from previous location where exception was thrown ---
```
Other tests within that build may have failed with a similar message, but they are not listed here. Check the link above for more info.

CC @Eilon,@muratg,@mkArtakMSFT

This issue was made automatically. If there is a problem contact @ryanbrandenburg.

### PR DESCRIPTION
**Pull request by halter73**
#2886 

This should also resolve #2776

### CODE REVIEW

**Comment by halter73** [comment]
Unfortunately this seems to be causing a bunch of new failures. Some of the failures would be best fixed with product changes we've made in the release/2.2 branch, but we're trying to avoid product changes in release/2.1 if it's only to fix tests. Ultimately, I think this change is too destabilizing for release/2.1 😢

File: src/Kestrel.Core/Properties/AssemblyInfo.cs [hunk 1]
```diff
@@ -6,6 +6,9 @@
...
+[assembly: InternalsVisibleTo("InMemory.FunctionalTests, PublicKey=0024000004800000940000000602000000240000525341310004000001000100f33a29044fa9d740c9b3213a93e57c84b472c84e0b8a0e1ae48e67a9f8f6de9d5f7f3d52ac23e48ac51801f1dc950abe901da34d2a9e3baadb141a17c77ef3c565dd5ee5054b91cf63bb3c6ab83f72ab3aafe93d0fc3c2348b764fafb0b1c0733de51459aeab46580384bf9d74c4e28164b7cde247f891ba07891c9d872ad2bb")]
+[assembly: InternalsVisibleTo("Sockets.BindTests, PublicKey=0024000004800000940000000602000000240000525341310004000001000100f33a29044fa9d740c9b3213a93e57c84b472c84e0b8a0e1ae48e67a9f8f6de9d5f7f3d52ac23e48ac51801f1dc950abe901da34d2a9e3baadb141a17c77ef3c565dd5ee5054b91cf63bb3c6ab83f72ab3aafe93d0fc3c2348b764fafb0b1c0733de51459aeab46580384bf9d74c4e28164b7cde247f891ba07891c9d872ad2bb")]
+[assembly: InternalsVisibleTo("Libuv.BindTests, PublicKey=0024000004800000940000000602000000240000525341310004000001000100f33a29044fa9d740c9b3213a93e57c84b472c84e0b8a0e1ae48e67a9f8f6de9d5f7f3d52ac23e48ac51801f1dc950abe901da34d2a9e3baadb141a17c77ef3c565dd5ee5054b91cf63bb3c6ab83f72ab3aafe93d0fc3c2348b764fafb0b1c0733de51459aeab46580384bf9d74c4e28164b7cde247f891ba07891c9d872ad2bb")]
```
Comment (Eilon) [inline src/Kestrel.Core/Properties/AssemblyInfo.cs]: 😭 Can we avoid *all* `src/*` changes?

**Comment by halter73** [comment]
We discussed this in today's Kestrel stand up. We decided that it isn't very feasible to make the in-memory functional test project reliable without making `src/*` changes similar to those we've made in 2.2.

**Comment by Eilon** [comment]
Could we move the tests into an existing assembly that has IVT on it already?

**Pull request closed without merging**